lib/datapacksjob.js

```diff
@@ -3016,12 +3018,68 @@ DataPacksJob.prototype.downloadPerformanceData = async function (jobInfo) {
 
+                let unique_omni_id = Array.from([...new Set(omni_id)]);
+
+                let res = [];
+
+                let elements = new Map();
+
+                let keys = ["preTransformBundle", "postTransformBundle", "sendJSONNode", "sendJSONPath", "responseJSONNode", "responseJSONPath", "useQueueableApexRemoting", "remoteTimeout",
+                    "sendJSONNode", "sendJSONPath", "responseJSONNode", "responseJSONPath", "cacheType", "chainOnStep", "failOnStepError", "scheduledJobId", "remoteOptions.configurationName",
+                    "bundle", "emailTemplateInformation.emailTemplateName", "restPath", "restMethod", "namedCredential", "type", "retryCount", "xmlPreTransformBundle", "xmlPostTransformBundle",
+                    "machineDeveloperName", "integrationProcedureKey", "useContinuation", "remoteOptions.useQueueable", "remoteOptions.useFuture", "remoteOptions.chainable", "remoteOptions.queueableChainable",
+                    "remoteOptions.matrixName", "Type", "Sub Type", "Language", "preIP", "postIP", "remoteClass", "remoteMethod"];
+
+                while (unique_omni_id.length > 0) {
+
+                    let chunk = unique_omni_id.splice(0, 150);
+                    let unique_omni_id_format = "";
+
+                    for (let j = 0; j < chunk.length; j++) {
+                        if (j == (chunk.length - 1)) {
+                            unique_omni_id_format = unique_omni_id_format + `'${chunk[j]}'`
+                        } else {
+                            unique_omni_id_format = unique_omni_id_format + `'${chunk[j]}',`;
+                        }
+                    }
+
+                    let additional_data = await self.vlocity.jsForceConnection.query(`SELECT ${self.vlocity.namespace}__PropertySet__c,Name,${self.vlocity.namespace}__Type__c,${self.vlocity.namespace}__OmniScriptId__c 
+                                          FROM ${self.vlocity.namespace}__Element__c WHERE ${self.vlocity.namespace}__OmniScriptId__c IN (${unique_omni_id_format})`);
+
+                    additional_data.records.map(item => {
+                        elements[item['Name'] + item[self.vlocity.namespace + '__OmniScriptId__c']] = item;
+                    })
+                }
+
+                records.map(item => {
+
+                    let element_data = JSON.parse(item[field_name])
+                    let element_name = element_data.ElementName;
+                    let element_omniScript_id = element_data.OmniScriptId;
+                    let key = element_name + element_omniScript_id;
+                    let propertyset = JSON.parse(elements[key][self.vlocity.namespace + '__PropertySet__c']);
+                    let element_type = elements[key][self.vlocity.namespace + '__Type__c'];
+
+                    for (let k in propertyset) {
+                        if (keys.includes(k) && (propertyset[k] !== "")) {
+                            element_data[k.substr(k.lastIndexOf('.') + 1)] = propertyset[k]
```

Comment:
This won't quite work for values with `.` 

If I do:

`var prop = {test: { test2 : "a"}};
console.log(prop['test.test2'])
`

I don't think it will print "a". I think it needs to traverse into the object by the path.